deepfence_ui/app/scripts/components/vulnerability-view/registry-scan/credentials-list.js

```diff
@@ -109,6 +134,278 @@ class RegistryCredentialsList extends React.PureComponent {
+        selectedDocIndex={selectedDocIndex}
+        onSubmit={(valuesIm) => {
+          const { scanType, scheduleScan, scheduleInterval,
+          scanRegistryType } = valuesIm.toJS();
+          const params = {
+            selectedDocIndex,
+            scanType,
+            scheduleScan,
+            scheduleInterval,
+            scanRegistryType
+          };
+          this.scanMultipleregistries(params);
+        }}
+      />
+    );
+  }
+
+  scanMultipleregistries(params) {
+    const { selectedDocIndex = {} } = params;
+    const indexes = Object.keys(selectedDocIndex);
+    indexes.forEach(index => {
+      this.scanRegistryImages(params, index);
+    })
+  }
+
+  triggerStartBulkScan(selectedDocIndex) {
+    const { clearScanContainerImageRegistryAction: clearAction } = this.props;
+    const modalProps = {
+      title: "Registry Vulnerability Scan",
+      modalContent: this.renderModalContent,
+      modalContentProps: {
+        selectedDocIndex,
+      },
+      contentStyles: {
+        width: "400px",
+      },
+      onHide: () => {
+        clearAction();
+      },
+    };
+    const { triggerModal } = this.props;
+    triggerModal("GENERIC_MODAL", modalProps);
+  }
+
+  triggerStopCVEScanModal(selectedDocIndex) {
+    const modalProps = {
+      dialogTitle: 'Stop Vulnerability Scan',
+      dialogBody: 'This will only stop the scans that are in queued state. It will not stop scans that are currently running. Do you want to continue ?',
+      confirmButtonText: 'Yes',
+      cancelButtonText: 'No',
+      onConfirmButtonClick: paramsIm => this.bulkStopCVEScan(selectedDocIndex, paramsIm),
+    };
+    const {triggerModal} = this.props;
+    return triggerModal('DIALOG_MODAL', modalProps);
+  }
+
+  scanRegistryImages(params) {
+    const {
+      registryType,
+      scanRegistryImagesAction: action,
+      resetSelection,
+      dataIm = Map(),
+      registryImageTagsValues,
+      priorityValue,
+    } = this.props;
+
+    const { selectedDocIndex, scanType, scheduleScan, scheduleInterval,
+      scanRegistryType } =
+      params;
+
+    const scanRegistryTypeValue = scanRegistryType && scanRegistryType === "scanAll" ? '' : scanRegistryType;
+    let priorityValueCheck = priorityValue && priorityValue[0] === 'priority' ? true : false;
+    scanRegistryType && scanRegistryType === "scanAll" ? priorityValueCheck = false : priorityValueCheck = priorityValueCheck;
+    const ImageTagsValues = registryImageTagsValues && registryImageTagsValues.map(el => el.value);
+    const data = dataIm.get(registryType, []);
+    const selectedDocIndexValues = Object.entries(selectedDocIndex);
+    const registryIdValue = selectedDocIndexValues.map(el => el[1].id);
+    const registryId = registryIdValue[0];
+    const schedule = scheduleScan && scheduleScan[0] ? true : false;
+    const scanAllImages = scanRegistryType && scanRegistryType == "scanAll" ? true : false;
+
+    if (schedule && scheduleInterval) {
```

Comment:
this if block can be simplified, if there is a schedule make its api call and if there is a scheduleInterval make its api call. we can remove a lot of duplication here